splice/web/api/content.py

```diff
@@ -0,0 +1,98 @@
+from flask import Blueprint, request
+from flask.json import jsonify
+from splice.web.api.content_upload import upload_signed_content, resign_content
+from splice.queries.common import session_scope
+from splice.queries.content import get_content, get_contents, insert_content, update_content
+
+
+content_bp = Blueprint('api.content', __name__, url_prefix='/api')
+
+
+@content_bp.route('/content/resignall', methods=['POST'])
+def handler_content_resign_all():
+    succeeded, failed = [], []
+    for content in get_contents():
+        try:
+            resign_content(content['name'], content['version'])
+        except Exception as e:
+            failed.append("%s: %s" % (content['name'], e))
+        else:
+            succeeded.append(content['name'])
+
+    return jsonify(succeeded=succeeded, failed=failed)
+
+
+@content_bp.route('/content/sign', methods=['POST'])
+def handler_content_upload():
+    """Signed and upload a piece of content to S3, return the URL if succeeds."""
+    def _is_allowed_file(name):
+        return '.' in name and name.rsplit('.', 1)[1].lower() in set(['zip'])
+
+    content_file = request.files["content"]
+    name = request.args.get('name')
+    version = request.args.get('version')
+
+    if not _is_allowed_file(content_file.filename):
+        return jsonify(message="Invalid content uploaded, expecting a zip file."), 400
+
+    if not name:
+        return jsonify(message="Missing content name in the query string."), 400
+
+    # try to fetch the content from the database
+    new_content = False
+    freeze = False
+    content_record = get_content(name)
+    if content_record is None:
+        content_record = {"name": name}
+        new_content = True
+
+    # determin the version based on the query string (if specified), and the current version in the database
+    if version:
+        try:
+            version = int(version)
```

Comment:
This is to re-sign a specific version, e.g. v1 gets some bug fixes to deploy in an a/b test. The user wants to keep the old version to finish the a/b test. Make sense?